src/main/java/jenkins/plugins/logstash/configuration/ElasticSearch.java

```diff
@@ -16,24 +17,27 @@
 
 public class ElasticSearch extends LogstashIndexer<ElasticSearchDao>
 {
-  protected String key;
-  protected String username;
-  protected Secret password;
+  private String username;
+  private Secret password;
+  private URI uri;
 
   @DataBoundConstructor
-  public ElasticSearch()
+  public ElasticSearch(String uri) throws URISyntaxException
```

Comment:
Why are we changing the constructor instead of using a `@DataBoundSetter`?

Reply:
When the uri is not valid it will fail to save the configuration when it is checked in the constructor.
When using a DataBoundSetter, you can save the globalconfiguration with